### PR TITLE
Prune deployment utilities

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Follows pattern set forth here: https://github.com/openshift/origin/pull/2260

Now for deployments.

Follow up PR will tie build and deployment prune utilities into ```osadm``` command.

/cc @abhgupta @smarterclayton @ironcladlou

### CODE REVIEW

File: pkg/deploy/prune/data.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+// DeploymentByDeploymentConfigIndexFunc indexes Deployment items by their associated DeploymentConfig, if none, index with key "orphan"
+func DeploymentByDeploymentConfigIndexFunc(obj interface{}) (string, error) {
+	controller, ok := obj.(*kapi.ReplicationController)
+	if !ok {
+		return "", fmt.Errorf("not a replication controller: %v", obj)
+	}
+	name, exists := controller.Annotations[deployapi.DeploymentConfigAnnotation]
```
Comment (abhgupta) [inline pkg/deploy/prune/data.go]: Change to --> deployutil.DeploymentConfigNameFor(controller)

Refer https://github.com/openshift/origin/pull/2227

File: pkg/deploy/prune/data.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	now := util.Now()
+	before := util.NewTime(now.Time.Add(-1 * d))
+	return func(item *kapi.ReplicationController) bool {
+		return item.CreationTimestamp.Before(before)
+	}
+}
+
+// FilterDeploymentsPredicate is a function that returns true if the replication controller is associated with a DeploymentConfig
+func FilterDeploymentsPredicate(item *kapi.ReplicationController) bool {
+	_, exists := item.Annotations[deployapi.DeploymentConfigAnnotation]
```
Comment (abhgupta) [inline pkg/deploy/prune/data.go]: Same here, use the pkg/deploy/util.go method.

File: pkg/deploy/prune/data.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	}
+
+	return &dataSet{
+		deploymentConfigStore: deploymentConfigStore,
+		deploymentIndexer:     deploymentIndexer,
+	}
+}
+
+func (d *dataSet) GetDeploymentConfig(controller *kapi.ReplicationController) (*deployapi.DeploymentConfig, bool, error) {
+	name, exists := controller.Annotations[deployapi.DeploymentConfigAnnotation]
```
Comment (abhgupta) [inline pkg/deploy/prune/data.go]: And here

**Comment by derekwaynecarr** [comment]
Updated to use deployutil to read Annotation values

File: pkg/deploy/prune/sort.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+func (s sortableReplicationControllers) Less(i, j int) bool {
+	return !s[i].CreationTimestamp.Before(s[j].CreationTimestamp)
+}
+
+func (s sortableReplicationControllers) Swap(i, j int) {
+	t := s[i]
+	s[i] = s[j]
+	s[j] = t
+}
```
Comment (abhgupta) [inline pkg/deploy/prune/sort.go]: Golang allows you to do:
```
s[i], s[j] = s[j], s[i]
```

**Comment by abhgupta** [comment]
LGTM

**Comment by derekwaynecarr** [comment]
[merge]

**Pull request merged**